modules/heat_conduction/src/kernels/ADMatHeatSource.C

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#include "ADMatHeatSource.h"
+
+registerADMooseObject("HeatConductionApp", ADMatHeatSource);
+
+defineADValidParams(
+    ADMatHeatSource,
+    ADKernel,
+    params.addParam<Real>("scalar", 1.0, "Scalar multiplied by the body force term");
+    params.addRequiredParam<MaterialPropertyName>("material",
+                                                  "Material multiplied by the body force term"););
```

Comment:
Modeling after the `BodyForce` kernel, perhaps this doc string should be "Material property describing the body force". If both `scalar` and the `_mat_prop` are multiplied by the body force term, then what is the body force term?

Reply:
I modeled this after the constant coefficient in `BodyForce`
```
params.addParam<Real>("value", 1.0, "Coefficient to multiply by the body force term");
```
I agree it sounds a bit weird. I'll change it to what you suggested